llvm/lib/SYCL/PrepareSYCLOpt.cpp

```diff
@@ -230,12 +231,48 @@ struct PrepareSYCLOptState {
     }
   }
 
+  void removeCallInstCasts(llvm::Module *M) {
```

Comment:
This PR is for a fix, so it might not have been obvious at the first place. :-)
What about a few comments on what is the problem, what does it does and why ?...

Reply:
I added a few comments